### ISSUE TITLE
[type-annotation-spacing] fails in object types under different conditions

### ISSUE THREAD

**Issue by zuzusik** [bradzacher/eslint-plugin-typescript#152]
rule config:
```json
"typescript/type-annotation-spacing": ["error", {"before": true, "after": true}]
```

code:
```ts
class Some {
  // fails on every `:` inside object typing (expected)
  a : {some: string, other: {more: number}}; 

  // doesn't fail (not expected)
  someMethod : (args : {some: string, other: {more: number}}) => void; 

  // doesn't fail (not expected)
  doSomething(args : {some: string, other: {more: number}}) : void {}
}

// doesn't fail (not expected)
function some(args : {some: string, other: {more: number}}) : void {} 

// doesn't fail (not expected)
const a : {some: string, other: {more: number}} = {some: '1', other: {more: 2}};
```

**Comment by bradzacher** [comment]
This is a problem with how `eslint` traverses the tree that `typescript-eslint-parser` provides: https://github.com/eslint/typescript-eslint-parser/issues/532#issuecomment-434172856
